src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -543,9 +547,60 @@ annotationExpression
         {#annotationExpression = #(#[EXPR,"EXPR"],#annotationExpression);}
     ;
 
+
+recordDeclaration![AST modifiers]
+    :   r:"record" id:id
+        (tp:typeParameters)?
+        rc:recordComponentsList
+        ic:implementsClause
+        rb:recordBodyDeclaration
+        {#recordDeclaration = #(#[RECORD_DEF, "RECORD_DEF"],
+                              modifiers, r, id, rc);}
+    ;
+
+// Build no AST for below rules until full records support
+recordComponentsList!
+    :   LPAREN recordComponents RPAREN
+    ;
+
+recordComponents!
+    // Taken from parameterDeclarationList
+    :   (   ( recordComponent )=> recordComponent
+            ( options {warnWhenFollowAmbig=false;} :
+                ( COMMA recordComponent ) => COMMA recordComponent )*
+            ( COMMA recordComponentVariableLength )?
+        |
+            recordComponentVariableLength
+        )?
+    ;
+
+recordComponentVariableLength!
+    :   parameterModifier t:variableLengthParameterTypeSpec ELLIPSIS id
+        declaratorBrackets[#t]
+    ;
+
+recordComponent!
+    :   annotations typeSpec[false] id
+    ;
+
+recordBodyDeclaration!
+    :   LCURLY
+        (   (compactConstructorDeclaration)=> compactConstructorDeclaration
+        |   field
+        |   SEMI
+        )*
+        RCURLY
+    ;
+
+compactConstructorDeclaration!
+    :    annotations modifiers id
+            constructorBody
+    ;
+//
```

Comment:
intentional?

Reply:
It was, to sort of group all of the rules related to #8267 together.  I've removed it.